tests/regression/lp-2032679/task.yaml

```diff
@@ -0,0 +1,34 @@
+summary: Refresh snap from 2.59 to stable having homedirs set
+
+systems: [ ubuntu-18*, ubuntu-2* ]
+
+prepare: |
+    apt remove -y snapd --purge
+    apt install -y snapd
+
+    mkdir -p /home/mydir/
+    useradd -m -d /home/mydir/mytest mytest
+    usermod -aG sudo mytest
+
+    tests.session -u mytest prepare
+
+restore: |
+    tests.session -u mytest restore
+    userdel mytest
+    rm -rf /home/mydir
+
+    # Restore the original snapd
+    apt remove -y snapd --purge
+    #shellcheck source=tests/lib/pkgdb.sh
+    . "$TESTSLIB/pkgdb.sh"
+    distro_install_build_snapd
+
+execute: |
+    snap install snapd --revision 19122
+    snap set system homedirs=/home/mydir/
+
+    snap install kubectl --classic
+    tests.session -u mytest exec sh -c "kubectl version" 2>&1 | NOMATCH "Permission denied"
+
```

Comment:
It might be a good idea to also test that 2.60.2 is failing as expected to double check that the test is really triggering the regression case.

```diff
diff --git a/tests/regression/lp-2032679/task.yaml b/tests/regression/lp-2032679/task.yaml
index 8cf82a4158..cc2b8bc7bc 100644
--- a/tests/regression/lp-2032679/task.yaml
+++ b/tests/regression/lp-2032679/task.yaml
@@ -30,5 +30,10 @@ execute: |
     snap install kubectl --classic
     tests.session -u mytest exec sh -c "kubectl version" 2>&1 | NOMATCH "Permission denied"
 
+    # 2.60.2 regression
+    snap refresh snapd --revision 19993
+    tests.session -u mytest exec sh -c "kubectl version" 2>&1 | MATCH "Permission denied"
+
+
     snap refresh snapd --stable
     tests.session -u mytest exec sh -c "kubectl version" 2>&1 | NOMATCH "Permission denied"
```

Reply:
added